File: api/SistemaPenal/Entities/Prisioneiro.cs
using SistemaPenal.Entities.Abstractions;

namespace SistemaPenal.Entities;

public class Prisioneiro : Pessoa
{
    public DateTime DiaDeChegada { get; set; }
    public DateTime DiaDeSaidaOriginal { get; set; }
    public DateTime DiaDeSaidaAtualizado { get; set; }
    public int ContadorDeLivros { get; set; }
    public int AnoAtual { get; set; }

    public List<Livro> Livros { get; set; } = [];
    public List<Estudo> Estudos { get; set; } = [];
    public List<DiaDeTrabalho> DiasDeTrabalho { get; set; } = [];
}

File: sistema-penal/src/models/Prisioneiro.ts
import { Pessoa } from "./abstract/Pessoa";
import type { DiaDeTrabalho } from "./DiaDeTrabalho";
import type { Estudo } from "./Estudo";
import type { Livro } from "./Livro";

export class Prisioneiro extends Pessoa {
  diaDeChegada: Date;
  diaDeSaidaOriginal: Date;
  diaDeSaidaAtualizado: Date;
  contadorDeLivros: number;
  anoAtual: number;
  livros: Livro[];
  estudos: Estudo[];
  diasDeTrabalho: DiaDeTrabalho[];

  constructor(
    nome: string,
    dataNascimento: Date,
    cpf: string,
    diaDeChegada: Date,
    diaDeSaidaOriginal: Date,
    diaDeSaidaAtualizado: Date,
    contadorDeLivros: number,
    anoAtual: number,
    livros: Livro[],
    estudos: Estudo[],
    diasDeTrabalho: DiaDeTrabalho[],
    id?: string,
  ) {
    super(nome, dataNascimento, cpf, id);
    this.diaDeChegada = diaDeChegada;
    this.diaDeSaidaOriginal = diaDeSaidaOriginal;
    this.diaDeSaidaAtualizado = diaDeSaidaAtualizado;
    this.contadorDeLivros = contadorDeLivros;
    this.anoAtual = anoAtual;
    this.livros = livros;
    this.estudos = estudos;
    this.diasDeTrabalho = diasDeTrabalho;
  }
}
